streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -425,6 +427,9 @@ public void flushInternal() {
     @Override
     public void close() {
         flush();
+        options.dispose();
+        wOptions.dispose();
+        fOptions.dispose();
```

Comment:
I think it's not sufficient if one calls `openDB` which is a public method. `isClosed` could work too, but it seems like it would affect more code (unless we only checked `openDB`). I'm a fan of final fields (as you know), but maybe making them not final and adding a comment about the lifecycle makes it clearer in this case.